salto-io/salto
# SALTO-3975 Deploy group schema for custom fields and  add validator for base fields

1. Created endpoint for groupSchema in the config file for modification.
2. We noticed that the user cant simply delete attribute from group schema because of Okta API demands.
3. So we created a filter that assign null value when the user delete a custom field in order for Okta to understand the field need to be deleted.
4. I also created change validator that checks that base attributes didn't change + test for the validator. 

---

_Additional context for reviewer_
To test the filter manually do as follows:
1. Delete a custom attribute from groupSchema instance. 
2. Do `salto deploy` and check that the changes occurred and that the nacl has no deleted field.
3. Try to add and modify custom fields as well. 

To test the validator manually, do as follows:
1. Change bae attribute in groupSchema instance. 
2. Do 'salto deploy --dry-run'
---
_Release Notes_: 
None
---
_User Notifications_: 
None
